esphome/components/voice_assistant/voice_assistant.cpp

```diff
@@ -483,8 +490,9 @@ void VoiceAssistant::on_event(const api::VoiceAssistantEventResponse &msg) {
     }
     case api::enums::VOICE_ASSISTANT_RUN_END: {
       ESP_LOGD(TAG, "Assist Pipeline ended");
-      if (this->state_ != State::STREAMING_RESPONSE && this->state_ != State::IDLE) {
-        this->set_state_(State::IDLE, State::IDLE);
+      if (this->state_ == State::STREAMING_MICROPHONE) {
+        // No need to stop the microphone since we didn't use the speaker
+        this->set_state_(State::WAITING_FOR_VAD, State::WAITING_FOR_VAD);
       }
```

Comment:
Gotta check wake word is even being used =)
```suggestion
      if (this->state_ == State::STREAMING_MICROPHONE) {
#ifdef USE_ESP_ADF
        if (this->use_wake_word_) {
          rb_reset(this->ring_buffer_);
          // No need to stop the microphone since we didn't use the speaker
          this->set_state_(State::WAIT_FOR_VAD, State::WAITING_FOR_VAD);
        } else
#else
        {
          this->set_state_(State::IDLE, State::IDLE);
        }
#endif
      }
```